doc/design.rst

```diff
@@ -580,20 +580,11 @@ The restic backup program guarantees the following:
 -  Accessing the unencrypted content of stored files and metadata should
    not be possible without a password for the repository. Everything
    except the metadata included for informational purposes in the key
-   files is encrypted and authenticated.
-
+   files is encrypted and authenticated. The cache is also encrypted by 
+   default to also prevent metadata leaks when encrypted filesystems are 
```

Comment:
I pulled most of this wording from @fd0's https://github.com/restic/restic/issues/208#issuecomment-246323673, but I'll admit I'm no expert on restic caching so I could be missing a thing or two. The cache isn't covered well in the [design doc](https://github.com/restic/restic/blob/master/doc/design.rst) and I'm not familiar enough with the code base to confidently answer this one. My assumption is the `encrypted filesystem` is referring to the encrypted parts of the repo that can be mounted via FUSE.

Regarding your first question, I believe you're correct that this could also apply more broadly.

Reply:
The cache is mostly a local copy of certain repository files. It basically contains a copy of the whole index and all packs containing tree blobs. The usage of the cache for the fuse mount does no really differ from the cache usage for all other commands. I'd just leave out the "when encrypted filesystems are mounted" part.

Actually "by default" is also wrong. There is currently no way to store a decrypted cache. So I think the important message here is just, that the cache only stores encrypted data which prevents metadata leaking from the cache.